PyInstaller/loader/pyi_importers.py

```diff
@@ -284,8 +284,9 @@ def load_module(self, fullname, path=None):
             # Release the interpreter's import lock.
             imp.release_lock()
 
-        # Module returned only in case of no exception.
-        return module
+        # Module returned only in case of no exception. In case the imported
+        # module has recreated itself, we need to re-read from sys.modules
+        return sys.modules[fullname]
```

Comment:
I'd prefer to have `module = sys.modules[fullname]` just behind the `exec()` in line 271. Because this is here the "problem" occurs.